### PR TITLE
changes for saving site details

### PR DESCRIPTION
**Pull request by midhun-aot**

### CODE REVIEW

File: backend/sites/src/app/dto/roles/role.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const CustomRoles = {
+    External: process.env.ROLES_EXTERNAL ?? 'site-external-user',
+    Internal: process.env.ROLES_INTERNAL ?? 'site-internal-user',
+    SiteRegistrar: process.env.ROLES_SITE_REGISTRAR ?? 'site-site-registrar',
```
Comment (nikhila-aot) [inline backend/sites/src/app/dto/roles/role.ts]: you can title this as site-registrar

File: backend/sites/src/app/dto/srApprovalStatus.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export enum SRApprovalStatusEnum
+{
```
Comment (nikhila-aot) [inline backend/sites/src/app/dto/srApprovalStatus.ts]: move this outside of dto (either to a common or utility folder)

File: backend/sites/src/app/entities/HistoryLog.entity.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { Field, ObjectType } from '@nestjs/graphql';
+import GraphQLJSON from 'graphql-type-json';
+import {
+  Entity,
```
Comment (nikhila-aot) [inline backend/sites/src/app/entities/HistoryLog.entity.ts]: rename the file to have camel casing

File: frontend/site-registry/src/app/components/navigation/dto/SideNav.ts [hunk 4]
```diff
@@ -38,40 +39,89 @@ const getSideBarNavList = () => {
...
-      '/',
-      [
-        new SideNav(1, 'Folios', false, foliosIcon, '/folios'),
-        new SideNav(2, 'Purchases', false, purchasesIcon, '/purchases'),
-      ],
-      false,
-    ),
-    new SideNav(5, 'Cart', false, cartIcon, '/site/cart', [], true),
-  ];
+  console.log('userRole in getSideBarNavList',userRole);
```
Comment (nikhila-aot) [inline frontend/site-registry/src/app/components/navigation/dto/SideNav.ts]: can you remove the console

File: frontend/site-registry/src/app/components/table/body/TableBody.tsx [hunk 5]
```diff
@@ -59,6 +59,12 @@ const TableBody: FC<TableBodyProps> = ({
   resetAllRowsSelectedEventFlag,
   removePageFromAllRowsSelected,
 }) => {
+
+  useEffect(()=>{
+    console.log(data)
```
Comment (nikhila-aot) [inline frontend/site-registry/src/app/components/table/body/TableBody.tsx]: remove the console

File: frontend/site-registry/src/app/helpers/utility.ts [hunk 6]
```diff
@@ -152,3 +152,72 @@ export const showNotification = (
...
+    const user = getUser();
+    if(user !== null)
+    {
+      const userRoles: any = user.profile?.role;
+      switch(roleType)
+      {        
+         case 'client':
+        
+        
+          if(userRoles.includes("formsflow-client") || userRoles.includes("site-external-user"))
```
Comment (nikhila-aot) [inline frontend/site-registry/src/app/helpers/utility.ts]: we have to discuss this - on having the formsflow client role

File: backend/sites/src/app/services/site/site.service.ts [hunk 7]
```diff
@@ -238,4 +271,120 @@ export class SiteService {
       throw new Error('Failed to retrieve site ids.');
     }
   }
+
+  async saveSiteDetails(inputDTO: SaveSiteDetailsDTO, userInfo: any): Promise<boolean> {
+
+    try
+    {
+    if (!inputDTO) {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/site/site.service.ts]: can we have the if else block

File: backend/sites/src/app/services/site/site.service.ts [hunk 8]
```diff
@@ -238,4 +271,120 @@ export class SiteService {
...
+            await transactionalEntityManager.save(Sites, sitesSummary);
+          } else {
+            console.log('No changes To Site Summary');
+          }
+
+          if (events) {
+
+            events.map(event=> event.srAction = SRApprovalStatusEnum.Pending);
+
+            transactionalEntityManager.save(Events, events);
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/site/site.service.ts]: no await here

File: backend/sites/src/app/services/site/site.service.ts [hunk 9]
```diff
@@ -238,4 +271,120 @@ export class SiteService {
...
+            userId : userInfo??userInfo.sub,
+            content : inputDTO,
+            id: null,
+            whoCreated: userInfo??userInfo.givenName,
+            whenCreated: new Date(),
+            whenUpdated : new Date(),
+            whoUpdated: userInfo??userInfo.givenName,
+          }
+
+          this.historyLogRepository.save(historyLog);
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/site/site.service.ts]: should we have an await? and assign result of await to a variable and handle it

File: backend/sites/src/app/services/site/site.service.ts [hunk 10]
```diff
@@ -1,14 +1,27 @@
 import { Injectable } from '@nestjs/common';
-import { InjectRepository } from '@nestjs/typeorm';
-import { Repository } from 'typeorm';
+import { InjectEntityManager, InjectRepository } from '@nestjs/typeorm';
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/site/site.service.ts:2]: include unit test class for the save site details

**Review by nikhila-aot — CHANGES_REQUESTED** [review]

File: backend/sites/src/app/dto/roles/role.ts (hunk 1 above)
Comment (midhun-aot) [reply backend/sites/src/app/dto/roles/role.ts]: Done

File: backend/sites/src/app/dto/srApprovalStatus.ts (hunk 2 above)
Comment (midhun-aot) [reply backend/sites/src/app/dto/srApprovalStatus.ts]: Done

File: frontend/site-registry/src/app/components/navigation/dto/SideNav.ts (hunk 4 above)
Comment (midhun-aot) [reply frontend/site-registry/src/app/components/navigation/dto/SideNav.ts]: Done

File: backend/sites/src/app/entities/HistoryLog.entity.ts (hunk 3 above)
Comment (midhun-aot) [reply backend/sites/src/app/entities/HistoryLog.entity.ts]: Done

File: frontend/site-registry/src/app/helpers/utility.ts (hunk 6 above)
Comment (midhun-aot) [reply frontend/site-registry/src/app/helpers/utility.ts]: Removed formsflow-client in code

File: frontend/site-registry/src/app/components/table/body/TableBody.tsx (hunk 5 above)
Comment (midhun-aot) [reply frontend/site-registry/src/app/components/table/body/TableBody.tsx]: done

File: backend/sites/src/app/services/site/site.service.ts (hunk 8 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/site/site.service.ts]: Done

File: backend/sites/src/app/services/site/site.service.ts (hunk 9 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/site/site.service.ts]: done added await

File: backend/sites/src/app/services/site/site.service.ts (hunk 7 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/site/site.service.ts]: done

File: backend/sites/src/app/dto/roles/role.ts [hunk 11]
```diff
@@ -1,5 +1,5 @@
 export const CustomRoles = {
```
Comment (nikhila-aot) [inline backend/sites/src/app/dto/roles/role.ts]: can we move this outside of the dto folder? app-> common or utility

File: backend/sites/src/app/entities/siteHistoryLog.entity.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+@Index('idx_historylog_user_id', ['userId'])
+export class HistoryLog extends BaseAuditEntity {
+  @Field()
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Field()
+  @Column('character varying', { name: 'user_id', length: 100 })
+  userId: string;
+
```
Comment (nikhila-aot) [inline backend/sites/src/app/entities/siteHistoryLog.entity.ts:23]: have siteid also as a column for this entity

File: backend/sites/src/app/entities/HistoryLog.entity.ts (hunk 3 above)
Comment (nikhila-aot) [reply backend/sites/src/app/entities/HistoryLog.entity.ts]: I do not see the change

File: backend/sites/src/app/entities/siteHistoryLog.entity.ts (hunk 12 above)
Comment (midhun-aot) [reply backend/sites/src/app/entities/siteHistoryLog.entity.ts:23]: done

File: backend/sites/src/app/dto/roles/role.ts (hunk 11 above)
Comment (midhun-aot) [reply backend/sites/src/app/dto/roles/role.ts]: done

File: backend/sites/src/app/resolvers/folio/folio.resolver.ts [hunk 13]
```diff
@@ -121,14 +122,15 @@ export class FolioResolver {
...
-  @Roles({ roles: ['site-admin'], mode: RoleMatchingMode.ANY })
+ @Roles({ roles: [CustomRoles.External], mode: RoleMatchingMode.ANY })
   @Mutation(() => FolioResponse, { name: 'addSiteToFolio' })
   async addSiteToFolio(
     @Args('folioDTO', { type: () => [FolioContentDTO] }, new ValidationPipe())
     folioContentDTO: [FolioContentDTO],
     @AuthenticatedUser()
     user: any,
   ) {
+    console.log("user",user)
```
Comment (nikhila-aot) [inline backend/sites/src/app/resolvers/folio/folio.resolver.ts]: remove console log

**Review by nikhila-aot — APPROVED** [review]

**Pull request merged**